lib/vanagon/component.rb

```diff
@@ -290,6 +290,10 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      { name => { 'version' => version, 'ref' => options[:ref] }.delete_if { |k, v| !v } }
```

Comment:
Optimize for human readability until this becomes a performance bottleneck.

There's still iteration, there's still conditionals.